### PR TITLE
[SeaTunnel #919] Add spark-hudi source plugin 

### LINKED ISSUE #919: [Feature][Plugin] spark-hudi source plugin
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

spark-hudi source plugin

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by simon824**
## Purpose of this pull request
Spark-hudi source plugin 
closes #919

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] Change does not need document change, or I will submit document change to https://github.com/apache/incubator-seatunnel-website later

### CODE REVIEW

File: seatunnel-connectors/seatunnel-connector-spark-hudi/src/main/scala/org/apache/seatunnel/spark/source/Hudi.scala [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+class Hudi extends SparkBatchSource {
+
+  override def prepare(env: SparkEnvironment): Unit = {}
+
+  override def checkConfig(): CheckResult = {
+    val requiredOptions = Seq("hoodie.datasource.read.paths")
+    var missingOptions = new StringBuilder
+    requiredOptions.map(opt =>
+      if (!config.hasPath(opt))
+        missingOptions.append(opt).append(","))
```
Comment (wuchunfu) [inline seatunnel-connectors/seatunnel-connector-spark-hudi/src/main/scala/org/apache/seatunnel/spark/source/Hudi.scala:35]: If block needs braces

File: docs/en/configuration/source-plugins/Hudi.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+Read data from Hudi.
+
+## Options
+
+| name           | type   | required | default value |
+| -------------- | ------ | -------- | ------------- |
+| [hoodie.datasource.read.paths](#hoodiedatasourcereadpaths-string)           | string | yes      | -             |
+
+Refer to [hudi read options](https://hudi.apache.org/docs/configurations/#Read-Options) for configurations.
```
Comment (RickyHuo) [inline docs/en/configuration/source-plugins/Hudi.md:13]: We can list `hoodie.*` in Options and show the reference below.

**Review by RickyHuo — CHANGES_REQUESTED** [review]

File: seatunnel-connectors/seatunnel-connector-spark-hudi/src/main/scala/org/apache/seatunnel/spark/source/Hudi.scala (hunk 1 above)
Comment (simon824) [reply seatunnel-connectors/seatunnel-connector-spark-hudi/src/main/scala/org/apache/seatunnel/spark/source/Hudi.scala]: Updated, thx.

File: docs/en/configuration/source-plugins/Hudi.md (hunk 2 above)
Comment (simon824) [reply docs/en/configuration/source-plugins/Hudi.md:23]: Updated, thx.

**Pull request merged**